### PR TITLE
fix :Added ZIO.readFile support for ScalaJS

### LINKED ISSUE #8295: Add ZIO.readFile support for ScalaJS

### PR DESCRIPTION
**Pull request by guptadeepak8**
fixes #8295
/claim #8295

### CODE REVIEW

File: core-tests/js/src/test/scala/zio/interop/ZIOPlatformSpecificTest.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import zio._
+import zio.test._
+import zio.test.Assertion._
+
+object ZIOPlatformSpecificTest extends DefaultRunnableSpec {
```
Comment (adamgfraser) [inline core-tests/js/src/test/scala/zio/interop/ZIOPlatformSpecificTest.scala]:
```suggestion
object ZIOPlatformSpecificSpec extends DefaultRunnableSpec {
```

File: core/js/src/main/scala/zio/ZIOPlatformSpecific.scala [hunk 2]
```diff
@@ -15,9 +15,11 @@
  */
 
 package zio
-
+import zio._
 import zio.stacktracer.TracingImplicits.disableAutoTrace
 
+import java.io.FileReader
+import org.scalajs.dom.Blob
```
Comment (adamgfraser) [inline core/js/src/main/scala/zio/ZIOPlatformSpecific.scala]:
```suggestion
```

File: core/js/src/main/scala/zio/ZIOPlatformSpecific.scala [hunk 3]
```diff
@@ -15,9 +15,11 @@
  */
 
 package zio
-
+import zio._
```
Comment (adamgfraser) [inline core/js/src/main/scala/zio/ZIOPlatformSpecific.scala]:
```suggestion

```

File: core/js/src/main/scala/zio/ZIOPlatformSpecific.scala [hunk 4]
```diff
@@ -35,6 +37,29 @@ private[zio] trait ZIOPlatformSpecific[-R, +E, +A] { self: ZIO[R, E, A] =>
...
+  /**
+   * Reads a file in the browser environment using FileReader API. This method
+   * uses the browser's File API.
+   *
+   * @param file
+   *   Input file to read
+   * @return
+   *   ZIO effect with the file content as a String
+   */
+  def readFile(file: js.File)(implicit trace: Trace): Task[String] =
```
Comment (adamgfraser) [inline core/js/src/main/scala/zio/ZIOPlatformSpecific.scala:51]: This should be a method on the companion object, not the trait. It is constructing a new workflow, not transforming an existing one.

**Review by adamgfraser — CHANGES_REQUESTED** [review]
Thanks for working on this! Left some comments.

**Comment by guptadeepak8** [comment]
@adamgfraser  i have made the changes

File: core/js/src/main/scala/zio/ZIOPlatformSpecific.scala [hunk 5]
```diff
@@ -18,6 +18,8 @@ package zio
 
 import zio.stacktracer.TracingImplicits.disableAutoTrace
 
+import java.io.FileReader
+
```
Comment (adamgfraser) [inline core/js/src/main/scala/zio/ZIOPlatformSpecific.scala]:
```suggestion
```

**Review by adamgfraser — CHANGES_REQUESTED** [review]
Can you format?

**Comment by guptadeepak8** [comment]
@adamgfraser made changes

**Comment by adamgfraser** [comment]
These changes to not address my previous review comments and the code does not appear to have ever been compiling or even to use valid operators. Closing.

**Pull request closed without merging**